### PR TITLE
Add --tmpfs to allow dirs to be mounted in tmpfs

### PR DESCRIPTION
**Pull request by crosbymichael**
By using `docker run --tmpfs /tmp` docker will mount the dir `/tmp` in tmpfs for the container.

### CODE REVIEW

**Comment by crosbymichael** [comment]
@LK4D4 I added an integration test for tmpfs mounts

**Comment by LK4D4** [comment]
@crosbymichael Thanks!

File: docs/sources/reference/api/docker_remote_api_v1.16.md [hunk 1]
```diff
@@ -153,7 +153,8 @@ Create a container
                "CapDrop": ["MKNOD"],
                "RestartPolicy": { "Name": "", "MaximumRetryCount": 0 },
                "NetworkMode": "bridge",
-               "Devices": []
+               "Devices": [],
+               "TmpfsMounts": []
```
Comment (LK4D4) [inline docs/sources/reference/api/docker_remote_api_v1.16.md]: Shouldn't this be in 1.17? Also description in hostconfig section is needed. Also I remember that somewhere should be "what's new" section.

**Comment by jfrazelle** [comment]
LGTM minus the docs comment above about the version :)

**Comment by crosbymichael** [comment]
I'll reopen when this is ready, going to reconsider the volume proposal and copy functionality

**Comment by alberts** [comment]
@crosbymichael Any thoughts on the next steps here? Lots of proposals flying around:

https://github.com/docker/docker/issues/9803
https://github.com/docker/docker/issues/9250
https://github.com/docker/docker/pull/8484

**Pull request closed without merging**